File: java/src/main/java/fr/sdecout/handson/rest/book/BookSearchResponseItem.java
package fr.sdecout.handson.rest.book;

import fr.sdecout.handson.rest.shared.AuthorField;
import fr.sdecout.handson.rest.shared.ValidIsbn;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;

import java.util.List;

public record BookSearchResponseItem(
        @NotNull @ValidIsbn String isbn,
        @NotBlank String title,
        @Valid @NotEmpty List<AuthorField> authors
) {
}

File: kotlin/src/main/kotlin/fr/sdecout/handson/rest/book/BookSearchResponseItem.kt
package fr.sdecout.handson.rest.book

import fr.sdecout.handson.rest.shared.AuthorField
import fr.sdecout.handson.rest.shared.ValidIsbn
import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.NotNull

data class BookSearchResponseItem(
    @field:NotNull @field:ValidIsbn val isbn: String?,
    @field:NotBlank val title: String?,
    @field:Valid @field:NotEmpty val authors: List<AuthorField>?,
)
